### PR TITLE
enh(scheduler): add some info on debug log to Keeping original host c…

### PR DESCRIPTION
**Pull request by jean-christophe81**
add grpc module
add cbd link option to publish symbols for modules

REFS: MON-12283

### CODE REVIEW

File: centreon-broker/grpc/test/connector.cc [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+// TEST_F(TcpConnector, Simple) {
+//   tcp::connector connector(test_addr, test_port, -1);
+//   std::shared_ptr<io::stream> io{connector.open()};
+
+//   std::shared_ptr<io::raw> data{new io::raw()};
+//   std::shared_ptr<io::data> data2{new io::raw()};
+
+//   data->append(std::string("PING\n"));
+//   io->write(data);
```
Comment (rem31) [inline centreon-broker/grpc/test/connector.cc:76]: why ?

**Review by rem31 — CHANGES_REQUESTED** [review]

**Review by bouda1 — APPROVED** [review]

File: centreon-broker/grpc/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,103 @@
+##
+## Copyright 2011 Centreon
```
Comment (bouda1) [inline centreon-broker/grpc/CMakeLists.txt:2]: 2022

File: centreon-broker/grpc/inc/com/centreon/broker/grpc/factory.hh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+#include "com/centreon/broker/namespace.hh"
+
+CCB_BEGIN()
+
+namespace grpc {
+/**
+ *  @class factory factory.hh "com/centreon/broker/grpc/factory.hh"
+ *  @brief TCP protocol factory.
+ *
+ *  Build TCP protocol objects.
```
Comment (bouda1) [inline centreon-broker/grpc/inc/com/centreon/broker/grpc/factory.hh:32]: Bad comment

File: centreon-broker/grpc/inc/com/centreon/broker/grpc/stream.hh [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+  int32_t flush() override;
+  int32_t stop() override;
+
+  bool is_down() const;
+
+  // template <class call_back_type>
+  // void set_write_callback(call_back_type&& new_callback) {
+  //   _channel->set_write_callback(new_callback);
+  // }
```
Comment (bouda1) [inline centreon-broker/grpc/inc/com/centreon/broker/grpc/stream.hh:60]: still needed?

File: centreon-broker/grpc/test/grpc_test_keys/ca_1234.crt [hunk 5]
```diff
@@ -0,0 +1,33 @@
+-----BEGIN CERTIFICATE-----
+MIIFpTCCA42gAwIBAgIUPvltjGPvcLx6j5/UlpoG70lUbZ4wDQYJKoZIhvcNAQEL
+BQAwYTELMAkGA1UEBhMCU1AxDzANBgNVBAgMBkZyYW5jZTERMA8GA1UEBwwIVG91
+bG91c2UxDTALBgNVBAoMBFRlc3QxDTALBgNVBAsMBFRlc3QxEDAOBgNVBAMMB1Jv
+b3QgQ0EwIBcNMjIwMzE1MTUxMzMwWhgPMjEyMjAyMTkxNTEzMzBaMGExCzAJBgNV
+BAYTAlNQMQ8wDQYDVQQIDAZGcmFuY2UxETAPBgNVBAcMCFRvdWxvdXNlMQ0wCwYD
```
Comment (bouda1) [inline centreon-broker/grpc/test/grpc_test_keys/ca_1234.crt:6]: Do you really need these certificates? Is it not possible to generate during the tests?

File: centreon-broker/tcp/src/connector.cc [hunk 6]
```diff
@@ -63,35 +63,22 @@ std::unique_ptr<io::stream> connector::open() {
...
+    return limit_endpoint::open();
+  }
+  catch (const std::exception &e) {
     log_v2::tcp()->debug(
-        "Unable to establish the connection to {}:{} (attempt {}): {}", _host,
-        _port, _is_ready_count, e.what());
-    return nullptr;
+    "Unable to establish the connection to {}:{} (attempt {}): {}", _host,
+    _port, _is_ready_count, e.what());
+    throw;
```
Comment (bouda1) [inline centreon-broker/tcp/src/connector.cc:72]: We returned nullptr and not you throw an exception. Is this ok for you ?

File: centreon-broker/test/test_util/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,38 @@
+##
+## Copyright 2011 Centreon
```
Comment (bouda1) [inline centreon-broker/test/test_util/CMakeLists.txt:2]: 2022

File: centreon-broker/tcp/src/connector.cc (hunk 6 above)
Comment (bouda1) [reply centreon-broker/tcp/src/connector.cc:72]: Maybe this leads to the same result... I don't know, so I ask :smile:

**Review by bouda1 — APPROVED** [review]

**Pull request merged**